### PR TITLE
Add data routes for test cycle initiation

### PR DESCRIPTION
**Pull request by spectranaut**
This PR is a draft because there is a lot of things I did experimentally and I want to run by @evmiguel. There are many questions in comments in the PR.

Additionally, at the high level, I think we need to decide on some norms related to route API. I added a bunch of end points, but I think we should have some consistency in their naming and use. Everything is currently a post request with data optionally in body. Should some of these be gets? Should we add data through the URL? 

```
# Original:

api/user                # saves a user
api/user/role           # saves a user to a role

# I added:

api/user/assigntorun    # saves a list of users to a run
api/user/removefromrun  # removes a list of users from a run
api/user/config         # gets all tester configurations (which tests can test qhich players)

api/cycle               # saves a cycle
api/cycle/cycles        # gets all cycles
api/cycle/testversions  # gets all the possible test version
```

I tested everything with a bunch of curls, for example:
```
curl --location --request POST 'localhost:5000/api/user/config' --header 'Content-Type: application/json' --data-raw '{}'

curl --location --request POST 'localhost:5000/api/cycle/testversions' --header 'Content-Type: application/json' --data-raw '{}'

curl --location --request POST 'localhost:5000/api/cycle' --header 'Content-Type: application/json' --data-raw '{"name": "a test cycle", "test_version_id": 1, "created_user_id": 1, "runs": [{"at_id": 1, "at_version": "12th", "browser_id": 1, "browser_version": "666", "apg_example_id": 1, "users": [1,2]}]}'

curl --location --request POST 'localhost:5000/api/cycle/delete' --header 'Content-Type: application/json' --data-raw '{"id": 17}'

curl --location --request POST 'localhost:5000/api/cycle/cycles' --header 'Content-Type: application/json' --data-raw '{}'

curl --location --request POST 'localhost:5000/api/user/assigntorun' --header 'Content-Type: application/json' --data-raw '{"users": [1], "run_id": 2}'

curl --location --request POST 'localhost:5000/api/user/removefromrun' --header 'Content-Type: application/json' --data-raw '{"users": [1], "run_id": 2}'
```

### CODE REVIEW

File: server/routes/users.js [hunk 1]
```diff
@@ -6,4 +6,11 @@ const router = Router();
 router.post('/', UsersController.addUser);
 router.post('/role', UsersController.addUserToRole);
 
+router.post('/assigntorun', UsersController.assignUsersToRun);
```
Comment (evmiguel) [inline server/routes/users.js:9]: This can be `router.post('/run', UsersController.assignUsersToRun)`

File: server/routes/users.js [hunk 2]
```diff
@@ -6,4 +6,11 @@ const router = Router();
...
+router.post('/removefromrun', UsersController.removeUsersFromRun);
```
Comment (evmiguel) [inline server/routes/users.js:10]: This can be `router.delete('/run', UsersController.removeUsersFromRun)`

**Comment by spectranaut** [comment]
Ok @evmiguel can you take a quick look? I cleaned everything up a lot! You will have to run the db migrations.

**Comment by spectranaut** [comment]
update list of post/gets/deletes after talking to Erika:
```
POST  api/cycle               # saves a cycle
GET   api/cycle/              # gets all cycles
GET   api/cycle/testversions  # gets all the possible test version

POST  api/user/run            # saves a list of users to a run
DELETE api/user/run           # removes a user from a run
GET   api/user/config         # gets all tester configurations (which tests can test qhich players)

PUT   api/user                # saves a user
POST  api/user/role           # saves a user to a role

```

**Pull request merged**